crates/criticalup-core/src/cache.rs

```diff
@@ -0,0 +1,159 @@
+use md5::Md5;
+use reqwest::header::HeaderValue;
+use reqwest::{IntoUrl, Request, StatusCode};
+use sha2::Digest;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use crate::download_server_client::Connectivity;
+use tokio::fs as tokio_fs;
+
+use criticaltrust::manifests::ReleaseArtifactFormat;
+
+use reqwest::header::AUTHORIZATION;
+
+use crate::download_server_client::{unexpected_status, DownloadServerClient};
```

Comment:
We don't want to use the `download_server_client` here. 
This module content shouldn't link back to `download_server_client`. It should be isolated and offer pure cache information.  I see that in some cases you need the client to get the `cache` folder. Pass the reference to the cache folder instead, making the functions in this module independent of the download server client